### DISCUSSION TITLE
Is there any work in progress for log messages encoded over sysex to allow some quick "println" debugging?

### DISCUSSION THREAD

**Discussion by soymonitus** [SynthstromAudible/DelugeFirmware#296] [Ideas]
I have seen some PRs going on with transferring the firmware over sysex. That's good to know that Deluge is adopting some sysex squema. And it will help development speed for us who don't have debuggers. But apart from that, is there any work in progress to send from the Deluge sysex ancoded strings?
From MacOS you have for example SySex Librarian which can record incoming sysex messages and when you open them it's easy to see the encoded text inside. For some quick tests of which value has some variable that you want to know during development, it can be quite useful
<img width="839" alt="image" src="https://github.com/SynthstromAudible/DelugeFirmware/assets/6472762/579081ce-d6c5-413a-ac89-125220e2c776">

**Comment by bfredl** [comment]
yes, a prototype was implemented here (schema in the commit message):
   https://github.com/SynthstromAudible/DelugeFirmware/commit/28128f75e55d143f09aa0a6ea5f65d9487e7ed33

If you have webmidi working you can use https://bfredl.github.io/delugeclient/app.html as a client app ("get debug" for the text messages)

**Reply by soymonitus** [reply]
Love this!!! I will try it as soon as i get to my Deluge
